### PR TITLE
WIP: Neutral armies

### PR DESCRIPTION
**Pull request by tttppp**

### CODE REVIEW

File: variants/neutralarmies/neutralarmies.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	"github.com/zond/godip/variants/classical"
+	"github.com/zond/godip/variants/classical/start"
+	"github.com/zond/godip/variants/common"
+	"github.com/zond/godip/variants/hundred"
+)
+
+func NeutralOrders(state state.State) (ret map[godip.Province]godip.Adjudicator) {
+	ret = map[godip.Province]godip.Adjudicator{}
+	if state.Phase().Type() == godip.Movement {
+
```
Comment (zond) [inline variants/neutralarmies/neutralarmies.go]: What's this?

File: state/state.go [hunk 2]
```diff
@@ -7,11 +7,12 @@ import (
 	"github.com/zond/godip"
 )
 
-func New(graph godip.Graph, phase godip.Phase, backupRule godip.BackupRule) *State {
+func New(graph godip.Graph, phase godip.Phase, backupRule godip.BackupRule, neutralOrders func(State) map[godip.Province]godip.Adjudicator) *State {
```
[the remark this answers is not in the document]
Comment (zond) [reply state/state.go]: Never mind, I saw that they actually build :)

**Comment by zond** [comment]
You decide, I happily leave variant decisions to you :D

File: variants/neutralarmies/neutralarmies.go (hunk 1 above)
Comment (tttppp) [reply variants/neutralarmies/neutralarmies.go]: Ah, this is when I thought Go didn't support switch statements.

File: state/state.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (tttppp) [reply state/state.go]: Yes, these just build and hold. I was origianlly thinking of doing this as a very primitive AI player, but it's actually a variant behavior. AI players would control non-neutral units.

**Comment by zond** [comment]
Looks good, anyway :)

File: variants/neutralarmies/neutralarmies.go (hunk 1 above)
Comment (tttppp) [reply variants/neutralarmies/neutralarmies.go]: This is now fixed in #40

**Pull request closed without merging**